features/step_definitions/navigation_steps.rb

```diff
@@ -34,6 +34,24 @@
     raise "Couldn't find the #{text} in webpage"
   end
 end
+
+When(/^I wait until I see the name of "([^"]*)", refreshing the page$/) do |target|
+  case target
```

Comment:
here,  we see that we should get rid of the var $minion_fullhostname and make it as an attribute of the object.

THen we could use 
```get_node(target) ```
```  step %(I wait until i see "#{node.fullhostname}" text, refreshing the page```